### PR TITLE
Options ordered alphabetically in filter menu [transactions view]

### PR DESCRIPTION
**Pull request by Aryankpoor**
[#8727](https://github.com/hackclub/hcb/pull/8727) might not have fully resolved #8689.
Also noticed some glitches after the push

image

## Here the order is not correct
image

This PR should be able to order user options alphabetically.

### CODE REVIEW

**Comment by Aryankpoor** [comment]
Note: This PR only fixes the sorting not the glitches seen in the attached image

**Comment by Aryankpoor** [comment]
Seems to be working fine on deployment
image

**Review by sampoder — CHANGES_REQUESTED** [review]
Thanks! For the organisers one, I'd prefer for us to do it in SQL. Here's what I got to work:

```sql
@event.organizer_positions.joins(:user).order(Arel.sql("CONCAT(preferred_name, full_name) ASC"))
```

We should do the same for tags, something like `@event.tags.order(label: :asc)`.

**Comment by Aryankpoor** [comment]
Done! Let me know if everything looks good. 
Also wanted to mention that this PR also sorts ```Type``` alphabetically

## Original View
image

## Updated view
image

File: app/views/events/_filter_menu.html.erb [hunk 1]
```diff
@@ -3,7 +3,7 @@
     type="button"
     aria-label="Add filter..."
     class="tooltipped tooltipped--s pop menu__toggle menu__toggle--arrowless overflow-visible z-10"
-    data-menu-target="toggle" data-action="menu#toggle click@document->menu#close keydown@document->menu#keydown">
+    data-menu-target="toggle" data-action="menu#toggle click@document->menu#close keydown@document->keydown">
```
Comment (sampoder) [inline app/views/events/_filter_menu.html.erb]: Any reason for this change?

File: app/views/events/_filter_menu.html.erb [hunk 2]
```diff
@@ -37,10 +36,9 @@
         </div>
       <% end %>
       <div data-tabs-target="tab" id="user">
-        <% @organizers.each do |position| %>
+        <% @event.organizer_positions.joins(:user).order(Arel.sql("CONCAT(users.preferred_name, users.full_name) ASC")).each do |position| %>
```
Comment (sampoder) [inline app/views/events/_filter_menu.html.erb]: Can this be moved into `EventsController`? There should already be an `@organizers =` line.

**Review by sampoder — COMMENTED** [review]
Two quick comments; thanks!

File: app/views/events/_filter_menu.html.erb (hunk 1 above)
Comment (Aryankpoor) [reply app/views/events/_filter_menu.html.erb]: Aah sorry, that should not have been removed. That was just me testing something out as an attempt to resolve the glitches I mentioned earlier. 
I have added that back in

File: app/views/events/_filter_menu.html.erb (hunk 2 above)
Comment (Aryankpoor) [reply app/views/events/_filter_menu.html.erb]: Done :+1:

Comment (Aryankpoor) [reply app/views/events/_filter_menu.html.erb]: I have kept it in ```filter_menu``` for now as moving it to EventsController seems to be returning an error in deployment

Comment (Aryankpoor) [reply app/views/events/_filter_menu.html.erb]: I tested out the deployment after making this change and the transaction page broke. I can't figure out why though

Comment (Aryankpoor) [reply app/views/events/_filter_menu.html.erb]: I was trying out something like ```@organizers = @event.organizer_positions.joins(:user).order(Arel.sql("CONCAT(users.preferred_name, users.full_name) ASC"))```

Comment (sampoder) [reply app/views/events/_filter_menu.html.erb]: [`b5c4da0` (#8738)](https://github.com/hackclub/hcb/pull/8738/commits/b5c4da02731a8dee2795d4229dbc79016a726dd2) seems to be working for me locally. I think it's good to merge, might be staging timing out.

**Review by sampoder — APPROVED** [review]
LGTM!

**Pull request merged**